spec/integration/hawkular_client_spec.rb

```diff
@@ -258,13 +312,13 @@ module Hawkular::Client::RSpec
               actual_data[:data] = data
             end
             on.failure do |error|
-              actual_data[:data] = {}
-              puts 'error callback was called, reason: ' + error.to_s
+              actual_data[:data] = error
             end
           end
 
           actual_data = wait_for actual_data
-          expect(actual_data['status']).to eq('OK')
+          expect(actual_data['status']).to eq('OK') unless @agent_immutable
```

Comment:
I was trying (as i did that [agent_immutable code-style](https://github.com/hawkular/hawkular-client-ruby/pull/186) part on a separate PR) to have a single tests that could cover either cases. 

When running the live tests we only have immutable agents (as we use docker for testing). But when developing local we could either have an immutable agent or a non immutable one (when running outside of docker).

Any suggestion?